src/common/XChainIntegrator.sol

```diff
@@ -198,46 +182,129 @@ abstract contract XChainIntegrator is Auth {
 	}
 
 	/*/////////////////////////////////////////////////////
-					Vault Management
+					Address book management
 	/////////////////////////////////////////////////////*/
 
 	function addVault(
 		address _vault,
 		uint16 _chainId,
+		uint16 _srcPostmanId,
+		uint16 _dstPostmanId,
```

Comment:
maybe it makes sense to store a single enum of PostmanType
then look up the postman via `PostmanTyp => chainId => address` mapping?
This will make tracking postmans easier

Reply:
Got it, liked this solution. Already made a new push with those changes. Please take a look and tell me about it.

Reply:
I like!